src/NzbDrone.Core/Music/AlbumService.cs

```diff
@@ -282,12 +282,22 @@ public void SetAlbumMonitored(int albumId, bool monitored)
             var album = _albumRepository.Get(albumId);
             _albumRepository.SetMonitoredFlat(album, monitored);
 
+            // publish album edited event so artist stats update
+            _eventAggregator.PublishEvent(new AlbumEditedEvent(album, album));
+
             _logger.Debug("Monitored flag for Album:{0} was set to {1}", albumId, monitored);
         }
 
         public void SetMonitored(IEnumerable<int> ids, bool monitored)
         {
             _albumRepository.SetMonitored(ids, monitored);
+
+            // publish album edited event so artist stats update
+            foreach (int id in ids)
+            {
+                var album = _albumRepository.Get(id);
```

Comment:
Instead of querying the db each time in the loop for each `albumId` can we move repo query outside of loop with `var albums = _albumRepository.Get(ids);` and just loop the events so only one query is made?